endpoints/data_objects/src/main.cpp

```diff
@@ -356,6 +356,100 @@ namespace
 		std::int64_t remaining_bytes_;
 	}; // incremental_read
 
+	class incremental_write : public std::enable_shared_from_this<incremental_write>
+	{
+	  public:
+		incremental_write(
+			irods::http::session_pointer_type& _sess_ptr,
+			unsigned int _http_version,
+			bool _http_keep_alive,
+			irods::http::connection_facade& _conn,
+			std::unique_ptr<io::client::native_transport>& _tp,
+			std::unique_ptr<io::odstream>& _out,
+			io::odstream* _out_ptr,
+			std::unique_ptr<irods::at_scope_exit<std::function<void()>>>& _mark_pw_stream_as_usable,
+			std::string& _buffer,
+			std::int64_t _remaining_bytes,
+			std::int64_t _internal_buffer_size,
+			bool _is_parallel_write)
```

Comment:
Consider passing via an input struct.

Reply:
We can revisit this at a later time. The current implementation works and it's unlikely that anyone will need to modify this chunk of code once merged.

I'll open an issue to revisit this at a later time.

Reply:
Created #326.